src/firmware/app/control/trajectory_planner.c

```diff
@@ -195,17 +78,34 @@ app_trajectory_planner_interpolateConstantPeriodPositionTrajectory(
                            variable_time_trajectory->trajectory_elements[i - 1].time) +
                 variable_time_trajectory->trajectory_elements[i - 1].position.y;
 
-            const float interpolated_speed =
-                slope_speed *
+            const float interpolated_theta =
+                slope_theta *
+                    (trajectory_time -
+                     variable_time_trajectory->trajectory_elements[i - 1].time) +
+                variable_time_trajectory->trajectory_elements[i - 1].orientation;
+
+            const float interpolated_angular_speed =
+                slope_angular_speed *
                     (trajectory_time -
                      variable_time_trajectory->trajectory_elements[i - 1].time) +
-                variable_time_trajectory->speed_profile[i - 1];
+                variable_time_trajectory->trajectory_elements[i - 1].angular_speed;
+
+            const float interpolated_linear_speed =
+                slope_linear_speed *
+                    (trajectory_time -
+                     variable_time_trajectory->trajectory_elements[i - 1].time) +
+                variable_time_trajectory->trajectory_elements[i - 1].linear_speed;
```

Comment:
I think we can shorten this a bit by adding:
```
PositionTrajectoryElement_t prev_element = variable_time_trajectory->trajectory_elements[i - 1];
const float dt = trajectory_time - prev_element.time;
```
and then using those to compute the interpolated values.

Reply:
I added the linear interpolation math function that should address this issue